File: SecondYear/semestrul4/MPP/examen_practic/Rezolvari/SUB3_2023/networking/src/main/java/org/networking/rpcprotocol/Request.java
package org.networking.rpcprotocol;

import java.io.Serializable;

public class Request implements Serializable {
    private RequestType type;
    private Object data;

    Request() {
    }

    public RequestType getType() {
        return type;
    }

    public Object getData() {
        return data;
    }

    @Override
    public String toString() {
        return "Request{" +
                "type=" + type +
                ", data=" + data +
                '}';
    }

    public static class Builder {
        private final Request request = new Request();

        public Builder type(RequestType type) {
            request.type(type);
            return this;
        }

        public Builder data(Object data) {
            request.data(data);
            return this;
        }

        public Request build() {
            return request;
        }
    }

    private void data(Object data) {
        this.data = data;
    }

    private void type(RequestType type) {
        this.type = type;
    }

}

File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Networking/rpcprotocol/Request.cs

using System.Text.Json.Serialization;

namespace Networking.rpcprotocol;

[Serializable]
public class Request
{
    [JsonPropertyName("type")]
    public RequestType Type { get;  set; }
    public object Data { get;  set; }

    public override string ToString()
    {
        return $"Request{{ type={Type}, data={Data} }}";
    }
}
